tools/dynamic_framework_slicer/dynamic_framework_slicer.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash 
+set -eu
+
+BINARIES+=()
+while [[ $# -gt 0 ]]; do
+  arg="$1"
+  case $arg in
+      --in)
+      readonly IN="$2"
+      shift
+      ;;
+      --out)
+      readonly OUT="$2"
+      shift
+      ;;
+      --realpath)
+      readonly REALPATH="$2"
+      shift
+      ;;
+      *)
+      BINARIES+=("$1")
+      ;;
+  esac
+  shift
+done
+
+# Strip out any unnecessary slices from embedded dynamic frameworks to save space
+
+for bin in $BINARIES; do
+    # Lets figure out what slices the app has
+    bin_slices=$(xcrun lipo -info "$bin" | cut -d: -f3 | sort)
+
+    if [ -f "$IN" ]; then
```

Comment:
use `[[` over `[` since we're in bash, that way it is internal and doesn't spawn `test` (also more consistent within  the file).